tests/gce_virtual_machine_test.py

```diff
@@ -267,6 +270,38 @@ def testImageProjectFlag(self):
       self.assertIn('--image-project bar',
                     ' '.join(issue_command.call_args[0][0]))
 
+  def testVmMetadataFlag(self):
+    with self._PatchCriticalObjects() as issue_command:
+      self._mocked_flags.gcp_instance_metadata = ['k1:v1', 'k2:v2']
+      self._mocked_flags.owner = 'test-owner'
+      vm_spec = gce_virtual_machine.GceVmSpec(
+          'test_vm_spec.GCP', self._mocked_flags, image='image',
+          machine_type='test_machine_type')
+      vm = gce_virtual_machine.GceVirtualMachine(vm_spec)
+      vm._Create()
+      self.assertEquals(issue_command.call_count, 1)
+      actual_metadata = re.compile('--metadata\s+(.*)(\s+--)?').search(
+          ' '.join(issue_command.call_args[0][0])).group(1)
+      self.assertIn('k1=v1', actual_metadata)
+      self.assertIn('k2=v2', actual_metadata)
+      # Assert that FLAGS.owner is honored and added to instance metadata.
+      self.assertIn('owner=test-owner', actual_metadata)
+
+  def testVmMetadataFromFileFlag(self):
+    with self._PatchCriticalObjects() as issue_command:
+      self._mocked_flags.gcp_instance_metadata_from_file = ['k1:p1', 'k2:p2']
```

Comment:
Could you add a test case or modify this one to have a value that demonstrates the comma, like `['k1:p1,k2:p2', 'k3:p3']`?